raft/raft.go

```diff
@@ -1270,7 +1270,10 @@ func (r *raft) restore(s pb.Snapshot) bool {
 	}
 
 	// The normal peer can't become learner.
-	if !r.isLearner {
+	//
+	// Both of `prs` and `learnerPrs` are empty means the peer is new created by
+	// conf change, in which case we should accept snapshots make it as learner.
+	if (r.prs.len() > 0 || r.learnerPrs.len() > 0) && !r.isLearner {
```

Comment:
basically you want to check if r is initialized? if r.initialized() && !r.isLeaner {...} is clearer.